### ISSUE TITLE
Improve S1125: Flexibility in opting in and out constant patterns

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#8178]
### The context
After [this issue](https://github.com/SonarSource/sonar-dotnet/issues/2619), reported in 2019, was implemented [earlier this year](https://github.com/SonarSource/sonar-dotnet/pull/7687), we have got a user request to relax the reporting in presence of [constant patterns](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/patterns#1123-constant-pattern): https://github.com/SonarSource/sonar-dotnet/issues/8147.

### The goal 
Provide the user with more flexibility, by allowing to opt in and out:
- `==`-based comparisons: `== true` and `== false`
- `is false` comparison
- `is true` comparison

**Comment by AronGreen** [comment]
It would also be nice if the rule is relaxed when the comparison involves checking the run-time type of a boxed value.

A slimmed down example from a WPF IValueConverter implementation where the `is true` pattern is reported as a code smell, but returning `value` is not equivalent.

```C#
public object Convert(object value)
{
	return value is true;
}
```

**Comment by WarBorg** [comment]
hello, any news when this will be done ? we are also struggling to remove the `!` and use `is false` whenever possible in our code but this rule kinda hinders us at the moment

**Comment by MaxFmi** [comment]
Having `is false` as a valid option sounds good to me.

**Comment by Tim-Pohlmann** [comment]
I increased the priority of this ticket. Maybe we can pick it up in a future hardening sprint.

**Comment by pavel-mikula-sonarsource** [comment]
Moved to [NET-1664](https://sonarsource.atlassian.net/browse/NET-1664)

[NET-1664]: https://sonarsource.atlassian.net/browse/NET-1664?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
